doc/source/rllib/rllib-checkpoints-and-exports.rst

```diff
@@ -0,0 +1,338 @@
+same config that you used for the original algo, and only then call the new
+Algorithm's ``restore()`` method, passing it the checkpoint directory:
+
+.. literalinclude:: ../../../rllib/examples/documentation/checkpoints_and_exports.py
+    :language: python
+    :start-after: __restore-from-algo-checkpoint-2-begin__
+    :end-before: __restore-from-algo-checkpoint-2-end__
+
+The above procedure used to be the only way of restoring an algo, however, it is more tedious
+than using the ``from_checkpoint()`` utility as it requires an extra step and you will
+have to keep your original config stored somewhere.
+
+
+Which Algorithm checkpoint versions can I use?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+RLlib uses simple checkpoint versions (for example "v0" or "v1") to figure
+out how to restore an Algorithm (or a :py:class:`~ray.rllib.policy.policy.Policy`;
+see below) from a given
+checkpoint directory.
+
+From Ray 2.1 on, you can find the checkpoint version written in the
+``checkpoint_version.txt`` file at the top-level of your checkpoint directory.
+RLlib does not use this file or information therein, it solely exists for the
+user's convenience.
+
+From Ray RLlib 2.0 and up, all checkpoint versions will be
+backward compatible, meaning some RLlib version 2.x will be able to
+handle any checkpoints created by RLlib 2.0 or any version up to 2.x.
+
+
+Multi-agent Algorithm checkpoints
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In case you are working with a multi-agent setup and have more than one
+:py:class:`~ray.rllib.policy.policy.Policy` to train inside your
+:py:class:`~ray.rllib.algorithms.algorithm.Algorithm`, you
+can create an :py:class:`~ray.rllib.algorithms.algorithm.Algorithm` checkpoint in the
+exact same way as described above and will find your individual
+:py:class:`~ray.rllib.policy.policy.Policy` checkpoints
+inside the sub-directory ``policies/``.
+
+For example:
+
+.. literalinclude:: ../../../rllib/examples/documentation/checkpoints_and_exports.py
+    :language: python
+    :start-after: __multi-agent-checkpoints-begin__
+    :end-before: __multi-agent-checkpoints-end__
+
+
+Assuming
+TODO: Restoring a multi-agent :py:class:`~ray.rllib.algorithms.algorithm.Algorithm`,
+but only with some subset of the original policies
+
+
+Policy checkpoints
+------------------
+
+We have already looked at the ``policies/`` sub-directory inside an
+:py:class:`~ray.rllib.algorithms.algorithm.Algorithm` checkpoint dir
+and learned that individual policies inside the
+:py:class:`~ray.rllib.algorithms.algorithm.Algorithm` store all their state
+information under their policy ID inside that sub-directory.
+Thus, we now have the entire picture of a checkpoint:
+
+.. code-block::
+
+    .
+    ..
+    .is_checkpoint
+    .tune_metadata
+
+    checkpoint_version.txt          # <- contains checkpoint version, e.g. "v1"
+    state.pkl                       # <- state of the Algorithm (excluding Policy states)
+
+    policies/
```

Comment:
I hope we won't change this structure for the foreseeable future. snippets like these (which I really like btw!) tend to outdate quickly without anyone noticing.

Reply:
Hmm, good point.
Maybe we can "pin" this structure via a (hidden in the docs) assert in our `example/documents/... ` script that gets run in the CI. the assert could make sure that some created checkpoint produces these files and the error message of the assert would hint to also change this structure in the docs.

Reply:
![Screenshot 2022-10-07 at 13 17 57](https://user-images.githubusercontent.com/3462566/194541363-7aacddca-c959-4543-b239-3ca4f081f04c.png)

it also seems to wrap a little strangely. maybe unident a little?

Reply:
Has less indent now.